girder/api/v1/user.py

```diff
@@ -411,3 +424,54 @@ def getUserDetails(self, user, params):
             'nFolders': self.model('user').countFolders(
                 user, filterUser=self.getCurrentUser(), level=AccessType.READ)
         }
+
+    @access.public
+    @loadmodel(model='user', force=True)
+    @describeRoute(
+        Description('Verify an email address using a token.')
+        .param('id', 'The user ID to check.', paramType='path')
+        .param('token', 'The token to check.')
+        .errorResponse('The token is invalid or expired.', 401)
+    )
+    def verifyEmail(self, user, params):
+        self.requireParams('token', params)
+        token = self.model('token').load(
+            params['token'], force=True, objectId=False, exc=True)
+        delta = (token['expires'] - datetime.datetime.utcnow()).total_seconds()
+        hasScope = self.model('token').hasScope(
+            token, TokenScope.EMAIL_VERIFICATION)
+
+        if token.get('userId') != user['_id'] or delta <= 0 or not hasScope:
```

Comment:
Not sure I've ever seen `any` used with a set of arbitrary boolean expressions, looks weird. Usually it's a list or a list comprehension.

    if any([token.get('userId') != user['_id'], delta <= 0, not hasScope]):

Also, I directly pulled that code from `checkTemporaryPassword` :)